src/EStimLibrary/Core/ReusableIdPool.cs

```diff
@@ -74,16 +86,24 @@ public SortedSet<int> FreeIds
     /// negative number is given. Can be changed after construction.</param>
     public ReusableIdPool(int baseId, int numIds)
     {
-        // Store the base ID. Property setter handles defaulting to 0.
-        this.BaseId = baseId;
+        // Ensure BaseId is not negative
+        this.BaseId = Math.Max(baseId, MIN_BASE_ID);
 
-        // Store the number of IDs in this pool. Property setter handles
-        // defaulting to 0.
-        this.NumIds = numIds;
+        // Calculate the maximum allowable NumIds
+        long maxAllowableNumIds = (long)int.MaxValue - this.BaseId + 1;
 
-        // Initialize set of used IDs to empty. This is the only set manually
-        // that is method-editable not auto-generated.
-        this.UsedIds = new();
+        // Adjust NumIds if necessary
+        if ((long)numIds > maxAllowableNumIds)
```

Comment:
Nice catch! Move this check to the NumIds property set() method so any time the value is attempted to be adjusted, this check is performed